### PR TITLE
feat: define base ntsc perimission types

### PR DESCRIPTION
**Pull request by hamidzr**
## Description

- define separate permissions to view, create, and terminate NSC. 

[DET-8674]

## Commentary (optional)

- tensorboard operations would be tied to view access for requested experiments

## Checklist
- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

[DET-8674]: https://determinedai.atlassian.net/browse/DET-8674?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: proto/src/determined/rbac/v1/rbac.proto [hunk 1]
```diff
@@ -76,6 +76,14 @@ enum PermissionType {
   // Ability to delete experiment.
   PERMISSION_TYPE_DELETE_EXPERIMENT = 2006;
 
+  /* TODO: update the comments. we delegate tensorboard to view experiment artifacts. */
+  // Ability to create nscs.
+  PERMISSION_TYPE_CREATE_NSC = 3001;
+  // Ability to view nsc's.
+  PERMISSION_TYPE_VIEW_NSC = 3002;
+  // Ability to update nsc and nsc's lifecycle.
+  PERMISSION_TYPE_UPDATE_NSC = 3003;
```
Comment (hamidzr) [inline proto/src/determined/rbac/v1/rbac.proto:85]: not sure if we need to separate out update from create.
anyone with create access should be able to delete ntsc

Comment (hamidzr) [reply proto/src/determined/rbac/v1/rbac.proto:85]: also `NSC` is vague we'd want to clarify that if we keep these together

Comment (eecsliu) [reply proto/src/determined/rbac/v1/rbac.proto:85]: that makes sense, though I can imagine someone being given delete permissions in order to clean things up - like an automated account that runs cleanup scripts monthly on old NSCs

Comment (hamidzr) [reply proto/src/determined/rbac/v1/rbac.proto:85]: makes sense. that could be part of the update? I'm not sure we need an update permission for ntsc. there is nothing to update other than to kill.  By delete do you mean kill?

**Review by eecsliu — APPROVED** [review]

**Comment by hamidzr** [comment]
migrations?

**Pull request merged**